File: HotPetServer/models/Conversation.js
const mongoose = require("mongoose")

const ConversationSchema = new mongoose.Schema(
    {
        lastMessage: {type: String, required: true},
        lastMessageDate: {type: Date, required: true},
        sender: {
            type: mongoose.Schema.Types.ObjectId,
            ref: "User",
            required: true
        },
        receiver: {
            type: mongoose.Schema.Types.ObjectId,
            ref: "User",
            required: true
        },
    },
    {
        timestamps: {currentTime: () => Date.now()},
    }
)
module.exports = mongoose.model("Conversation", ConversationSchema)

File: app/src/main/java/com/example/hotpet/api/models/Conversation.kt
package com.example.hotpet.api.models

import java.io.Serializable
import java.util.*

data class Conversation(
    val _id: String?,
    val lastMessage: String,
    val lastMessageDate: Date,
    val sender: User?,
    val receiver: User?
) : Serializable

